components/support/rc_crypto/nss/src/pbkdf2.rs

```diff
@@ -0,0 +1,131 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::error::*;
+use crate::util::ensure_nss_initialized;
+use nss_sys::SECOidTag;
+
+// ***** BASED ON THE FOLLOWING IMPLEMENTATION *****
+// https://github.com/sqlcipher/sqlcipher/blob/4a81bea61e1da6fec222d713852830f1fd01aed2/src/crypto_nss.c#L182-L232
+// Which I believe is based on how it is done on WebCrypto:
+// https://searchfox.org/mozilla-central/rev/8ccea36c4fb09412609fb738c722830d7098602b/dom/crypto/WebCryptoTask.cpp#2567
+
+pub fn nss_pbkdf2_generate(
+    password: &mut [u8],
+    salt: &mut [u8],
+    iterations: u32,
+    out: &mut [u8],
+) -> Result<()> {
+    ensure_nss_initialized();
+    // Todo: Maybe add support for different hashing algorithms?
```

Comment:
Agree, how would you feel about adding an `Algorithm` param for this function?

Reply:
Done! (Used HashAlgorithm from context, not sure if that's the right choice, but seemed handy)